js/plugins/mcp/README.md

```diff
@@ -13,53 +15,99 @@ To get started, you'll need Genkit and the MCP plugin:
+
 Most MCP servers are built to run as spawned processes on the same machine using the `stdio` transport. When you supply the `serverProcess` option, you are specifying the command, arguments, and environment variables for spawning the server as a subprocess.
 
-### mcpClient() Options
+### [Deprecated] `mcpClient()`
+
+Note: This method is deprecated, please use `createMcpClient` or `createMcpManager` instead.
 
-- **`name`**: (required) The name for this client, which namespaces its tools and prompts.
-- **`version`**: (optional) The client's version number. Defaults to "1.0.0".
-- You must supply one of:
-  - **`serverProcess`**: Parameters for launching a local server process using the stdio MCP transport.
-    - **`command`**: Shell command path for launching the MCP server. Can be e.g. `npx` or `uvx` to download and run the server from a package manager.
-    - **`args`**: (optional) Array of string arguments to pass to the command.
-    - **`env`**: (optional) Key value object of environment variables to pass to the command.
-  - **`serverUrl`**: The URL of a remote server to connect to using the SSE MCP transport.
-  - **`serverWebsocketUrl`: The URL of a remote server to connect to using the WebSocket MCP transport.
-  - **`transport`**: An existing MCP transport object for connecting to the server.
-- **`rawToolResponses`**: (optional) A boolean flag. If `true`, tool responses are returned in their raw MCP format; otherwise, they are processed for Genkit compatibility.
+For simpler scenarios involving a single MCP server, or for backward compatibility, the legacy `mcpClient()` function is still available:
+
+```ts
+import { mcpClient } from 'genkitx-mcp/legacy'; // Note the import path
+
+const legacyFilesystemClient = mcpClient({
+  name: 'filesystemLegacy',
+  serverProcess: { /* ... */ },
+});
+```
+This function takes similar options to a single server configuration within `createMcpManager` (e.g., `name`, `version`, `serverProcess`, `rawToolResponses`) and directly returns a Genkit plugin for that single client. It is recommended to use `createMcpManager` for new projects.
 
 ### Using MCP Actions
 
-The Genkit MCP client automatically discovers available tools and prompts and registers them with Genkit making them available anywhere other tools and prompts can be used. To access resources, special `list_resources` and `read_resource` tools are registered that will access resources for the server.
+The Genkit MCP client manager, through its `getActiveTools()` method, discovers available tools from each connected and enabled server. These tools can then be provided to Genkit models for use in generation requests. Resources and MCP Tools are both wrapped as GenkitTools. 
 
-All MCP actions are namespaced under the name you supply, so a client called `filesystem` will register tools such as `filesystem/read_file`.
+<!-- TODO: Note about MCP prompts. -->
```

Comment:
@mbleigh @pavelgj 

How should we approach prompts with the newer dynamic API? Tools and resources can be wrapped as dynamic tools, but prompts should be treated differently, I suppose....

We don't have dynamic prompt support today and I'm wondering how to expose them in this new iteration of the plugin.

Should we also provide the user to install the MCP client as a plugin like we did for `mcpClient({..})` and provide prompts there?

Reply:
Let's think short-term / long-term.

Short-term, maybe we should essentially follow the API of `ExecutablePrompt` but not actually register them? So basically:

```ts
const mcpPrompt = mcpManager.prompt('some-prompt');
await mcpPrompt(input, options);
await mcpPrompt.stream(input, options);
await mcpPrompt.render(input, options);
```

and under the hood that's just rendering the prompt to `MessageData[]` and then either executing it, streaming it, or returning the rendered options?

Long-term I think we should have an MCP manager able to be used as a plugin with full dynamic registry integration.

Reply:
I like the idea of bespoke prompt references. One thing I'd suggest for the short term is to use a serverName qualifer to narrow it down:

```
const mcpPrompt = mcpManager.prompt({serverName: 'my-mcp-server', prompt: 'some-prompt'});
```

We can add support for dynamic look-ups as we go forward and handle server resolution when there is not conflicts.

> Long-term I think we should have an MCP manager able to be used as a plugin with full dynamic registry integration.

Agreed. We need to discuss more on how to support dynamic registrations via plugins.

Reply:
Maybe `mcpManager.prompt('server-name', 'some-prompt')`?

Reply:
Updated prompts.